src/Components/WebAssembly/testassets/ThreadingApp/Pages/Counter.razor

```diff
@@ -0,0 +1,79 @@
+﻿@page "/counter"
+@using System.Runtime.InteropServices
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        if(!OperatingSystem.IsBrowser())
+        {
+            return;
+        }
+
+        if (Thread.CurrentThread.ManagedThreadId != 1)
+        {
+            throw new Exception("We should be on main thread!");
+        }
+
+        Exception exc = null;
+        try
+        {
+            // send me to the thread pool
+            await Task.Delay(10).ConfigureAwait(false);
+            StateHasChanged(); // render should throw
+        }
+        catch(Exception ex)
+        {
+            exc=ex;
+            Console.WriteLine(ex.Message);
+        }
+        if (exc == null || exc.Message != "The current thread is not associated with the Dispatcher. Use InvokeAsync() to switch execution to the Dispatcher when triggering rendering or component state.")
+        {
+            throw new Exception("We should have thrown here!");
+        }
+
+        // test that we could create new thread
+        var tcs = new TaskCompletionSource<int>();
+        var t = new Thread(() => {
+            tcs.SetResult(Thread.CurrentThread.ManagedThreadId);
+        });
+        t.Start();
+        var newThreadId = await tcs.Task;
+        if (newThreadId == 1){
+            throw new Exception("We should be on new thread in the callback!");
+        }
+
+        new Timer(async (state) =>
```

Comment:
isn't this timer going to get randomly finalized since it's not being stored in a field? i'd prefer to see it get rooted